app/src/main/java/com/amaze/filemanager/filesystem/Operations.java

```diff
@@ -441,6 +441,61 @@ public static void rename(
 
       private final DataUtils dataUtils = DataUtils.getInstance();
 
+      private final Boolean isCaseSensitiveRename =
```

Comment:
Why is the case sensitive rename different?

Reply:
This approach only cares about the value changed and not the type of filesystem. Although it does unnecessary double renaming in case-sensitive filesystem, it doesn't need to check the type of filesystem. 

The 'localRename' function (can't think of a better name) is extracted from the original code to avoid code duplication.

Reply:
> Why is the case sensitive rename different? Shouldn't there just be filesystems that support case sensitive renames and filesystems that do not? And the renames be the same function?

I think this should be okay as we know for sure most file systems that android supports doesn't support case sensitivity (exception ext4 in which case it needs to be kernel enabled). But I think it would make a marginal performance improvement if second part of the AND condition in this is used first.

Reply:
I added comment on the purpose of isCaseSensitiveRename variable and refactored double rename part. Hopefully it is clearer.